e2e/cypress/integration/add_user_to_channel_from_profile_popover/profile_popover_spec.js

```diff
@@ -88,4 +77,465 @@ describe('Profile popover', () => {
+        cy.findByTestId('all_users-private_channel-checkbox').scrollIntoView().should('be.visible').click();
+        cy.findByTestId('channel_admin-public_channel-checkbox').scrollIntoView().should('be.visible').click();
+
+        // * Verify that all the sub-checkboxes are enabled.
+        verifyPermissionSubSections('channel_admin', 'public', true);
+
+        // # Clicking twice to disable it
+        cy.findByTestId('channel_admin-private_channel-checkbox').scrollIntoView().should('be.visible').click().click();
+
+        // * Verify that all the sub-checkboxes are disabled
+        verifyPermissionSubSections('channel_admin', 'private', false);
+
+        cy.findByTestId('saveSetting').should('be.visible').click();
+
+        // # Re-add the kicked out user
+        cy.apiAddUserToTeam(testTeam.id, testUser.id);
+        cy.apiLogout();
+
+        // # Login
+        cy.apiLogin(otherUser);
+        cy.apiAddUserToChannel(testChannel.id, otherUser.id);
+
+        // # Promote to channel admin
+        promoteToChannelOrTeamAdmin(otherUser, testChannel.id);
+        cy.visit(`/${testTeam.name}/channels/town-square`);
+
+        clickAddToChannel(testUser);
+
+        cy.get('div[aria-labelledby="addChannelModalLabel"]').within(() => {
+            // # Type "Channel" and press enter.
+            cy.get('input').should('be.visible').type('Channel').wait(TIMEOUTS.HALF_SEC).type('{enter}');
+
+            // * And verify that button is enabled
+            cy.get('#add-user-to-channel-modal__add-button').should('not.be.disabled');
+
+            // # Clear text box, type "private" and press enter.
+            cy.get('input').should('be.visible').clear().type('private').wait(TIMEOUTS.HALF_SEC).type('{enter}');
+
+            // * Verify that button is disabled.
+            cy.get('#add-user-to-channel-modal__add-button').should('be.disabled');
+
+            // # Clear text box, type "Channel" and press enter.
+            cy.get('input').clear().type('Channel').wait(TIMEOUTS.HALF_SEC).type('{enter}');
+
+            // # Now click the Add button
+            cy.get('#add-user-to-channel-modal__add-button').click();
+        });
+
+        // * Now verify that popup is gone
+        cy.get('div[aria-labelledby="addChannelModalLabel"]').should('not.be.visible');
+
+        // # Visit that channel
+        cy.visit(`/${testTeam.name}/channels/${testChannel.name}`);
+
+        // * Verify that user added message is there.
+        cy.findByTestId('postView').find('.post-message__text').should('contain.text', `@${testUser.username} added to the channel by you.`);
+    });
+
+    it('MM-T7 Add User — Team admins (Private only)', () => {
+        // # Do the system console setup
+        cy.apiAdminLogin();
+        cy.visit('/admin_console/user_management/permissions/system_scheme');
+        cy.findByTestId('all_users-public_channel-checkbox').scrollIntoView().should('be.visible').click();
```

Comment:
Same here for another click to fully uncheck.

Reply:
done